### ISSUE TITLE
Default -f option not working with every case.

### ISSUE THREAD

**Issue by Zulux91** [w0d4/yt-backup#37]
This I don't think it's a bug in itself since it's more a config issue, but there's an enhancement to be done, if considered as such.

**Describe the bug**
The default -f option included in the config.json gets stuck with heavy or long videos and doesn't proceed to download the video asked.

**To Reproduce**
Please share at least the video ID and your naming-format option.
Download the video with the id `eSXGuGm8s5w` and use `((571/272/402/337/315/313/401/336/308/400/271/335/303/299/399/137/248/334/302/298/398/247/136/333/244/135/397/332/243/134/396/331/242/133/395/330/160/394/278)[protocol!=http_dash_segments])+(bestaudio[acodec=opus]/bestaudio[protocol!=http_dash_segments])/best` in the video format options. It'll get stuck.

**Expected behavior**
A self-check to make sure the current video is actually being downloaded and/or auto switch to a fail-safe option `bestvideo+bestaudio/best` for instance.

**Debug log**
If applicable, add debug log of your problem with --debug
N/A
**Additional context**
N/A

**Comment by w0d4** [comment]
I will accept this as a bug first. I will try to reproduce later this day.

**Comment by w0d4** [comment]
Until now, I cannot reproduce this with any video. Also not with your provided videoID.
Your provided video downloaded fine in around 1.5 minutes with the default options.
If you try it with plain youtube-dl and the same settings, are there any error messages I can parse with my script and work around the issue?

**Comment by coletdjnz** [comment]
I think I have been having this same issue with this format selection (although in my own script), so I'll chime in with some info I've found.

For some videos they will begin to download then fail, or not download at all.
Looking at youtube-dl issues, the devs seem to point this is a server side issue with YouTube. [ytdl-org/youtube-dl/issues/21660](https://github.com/ytdl-org/youtube-dl/issues/21660)
Last comment on this (older) issue seems to suggest it's server side as well [ytdl-org/youtube-dl/issues/21854](https://github.com/ytdl-org/youtube-dl/issues/21854) 

In my case, using `bestvideo+bestaudio/best` seemed to fix this issue. 

**I did some testing:**
This video  seems to be failing on format 22. I have a few others like this too, same failing format. It seems like some formats are "broken", which this format selection tries to choose [ytdl-org/youtube-dl/issues/25633](https://github.com/ytdl-org/youtube-dl/issues/25633)
It seems to be an old issue - good detail here - [ytdl-org/youtube-dl/issues/17148](https://github.com/ytdl-org/youtube-dl/issues/17148)

(Also side note: the video @Zulux91 mentioned is working on format 22 for me)

**Case cxEg-30g75w:**
```
youtube-dl cxEg-30g75w -f "((571/272/402/337/315/313/401/336/308/400/271/335/303/299/399/137/248/334/302/298/398/247/136/333/244/135/397/332/243/134/396/331/242/133/395/330/160/394/278)[protocol!=http_dash_segments])+(bestaudio[acodec=opus]/bestaudio[protocol!=http_dash_segments])/best" --verbose -o "%(title)s_%(format)s_%(format_id)s.%(ext)s"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['cxEg-30g75w', '-f', '((571/272/402/337/315/313/401/336/308/400/271/335/303/299/399/137/248/334/302/298/398/247/136/333/244/135/397/332/243/134/396/331/242/133/395/330/160/394/278)[protocol!=http_dash_segments])+(bestaudio[acodec=opus]/bestaudio[protocol!=http_dash_segments])/best', '--verbose', '-o', '%(title)s_%(format)s_%(format_id)s.%(ext)s']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.07.28
[debug] Python version 3.8.3 (CPython) - Linux-5.7.9-1-MANJARO-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] cxEg-30g75w: Downloading webpage
[youtube] cxEg-30g75w: Downloading MPD manifest
[debug] Invoking downloader on 'https://r6---sn-8vq54voxo1-53a6.googlevideo.com/videoplayback?expire=1598277338&ei=enJDX9j1JJfoowPi1r-IAQ&ip=REDACTED&id=o-AGMK1P89LaH2CGXizmDak5d7UbDdLbE6gpWxDAh6a37a&itag=22&source=youtube&requiressl=yes&mh=B6&mm=31%2C26&mn=sn-8vq54voxo1-53a6%2Csn-a5mlrnes&ms=au%2Conr&mv=m&mvi=6&pl=22&initcwndbps=1610000&vprv=1&mime=video%2Fmp4&ratebypass=yes&dur=313.562&lmt=1498737529659884&mt=1598255558&fvip=6&beids=9466588&c=WEB&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAOpHXMKr2jInwt6eLdcFfuZAYZbQRdExTPruFFMzyzpGAiAwvDiL3C-V0phV21BvutuzdxFz958X-C8HHyLylCUZnw%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgTv4WCAAgu-exJ0k8fVMDhV3lmjI_PiRnoTzDiJY38ysCIQCreqlACcJ8ij5KtYYKAgSqRK_EdzyzJYY8jhIP5HwuiQ%3D%3D'
[download] Destination: MrSpeedrun Owns Annoying Youngster Bluepointed_22 - 1280x720 (720p)_22.mp4
[download]   0.4% of 25.18MiB at 179.39KiB/s ETA 02:23[download] Got server HTTP error: Downloaded 102501 bytes, expected 26404906 bytes. Retrying (attempt 1 of 10)...


ERROR: Did not get any data blocks
  File "/usr/bin/youtube-dl", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3.8/site-packages/youtube_dl/__init__.py", line 474, in main
    _real_main(argv)
  File "/usr/lib/python3.8/site-packages/youtube_dl/__init__.py", line 464, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 2018, in download
    res = self.extract_info(
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 808, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 863, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 1644, in process_video_result
    self.process_info(new_info)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 1926, in process_info
    success = dl(filename, info_dict)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 1865, in dl
    return fd.download(name, info)
  File "/usr/lib/python3.8/site-packages/youtube_dl/downloader/common.py", line 366, in download
    return self.real_download(filename, info_dict)
  File "/usr/lib/python3.8/site-packages/youtube_dl/downloader/http.py", line 342, in real_download
    return download()
  File "/usr/lib/python3.8/site-packages/youtube_dl/downloader/http.py", line 312, in download
    self.report_error('Did not get any data blocks')
  File "/usr/lib/python3.8/site-packages/youtube_dl/downloader/common.py", line 165, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 625, in report_error
    self.trouble(error_message, tb)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 587, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())

```
It seems for be format 22 thats causing the issue for this one:
```
youtube-dl cxEg-30g75w -f 22
[youtube] cxEg-30g75w: Downloading webpage
[youtube] cxEg-30g75w: Downloading MPD manifest
[download] Destination: MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.mp4
[download]   0.4% of 25.18MiB at 177.77KiB/s ETA 02:24[download] Got server HTTP error: Downloaded 102501 bytes, expected 26404906 bytes. Retrying (attempt 1 of 10)...


ERROR: Did not get any data blocks
```


Using `bestvideo+bestaudio/best` it works:

```
youtube-dl cxEg-30g75w -f "bestvideo+bestaudio/best"
[youtube] cxEg-30g75w: Downloading webpage
[youtube] cxEg-30g75w: Downloading MPD manifest
WARNING: Requested formats are incompatible for merge and will be merged into mkv.
[dashsegments] Total fragments: 64
[download] Destination: MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.f299.mp4
[download] 100% of 91.33MiB in 00:12
[dashsegments] Total fragments: 33
[download] Destination: MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.f251.webm
[download] 100% of 4.83MiB in 00:35
[ffmpeg] Merging formats into "MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.mkv"
Deleting original file MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.f299.mp4 (pass -k to keep)
Deleting original file MrSpeedrun Owns Annoying Youngster Bluepointed-cxEg-30g75w.f251.webm (pass -k to keep)
```

I tested with using a VPN to a different country to connect to different servers and I still got the same error.

**Comment by Zulux91** [comment]
Yep, that's the way I leave it: `bestvideo+bestaudio/best`. This works flawlessly for me and that's how I'm currently using it.

**Comment by w0d4** [comment]
I assume this to be solved with 0308101
